src/ui/DynamicFacetManager/DynamicFacetManager.ts

```diff
@@ -136,12 +155,12 @@ export class DynamicFacetManager extends Component {
   }
 
   private isDynamicFacet(component: Component) {
-    return component instanceof DynamicFacet;
+    return DynamicFacetManager.allCompatibleFacets.indexOf(component.type) !== -1;
   }
```

Comment:
Instead of redeclaring a list of components that implement `IDynamicFacet`, would it be better to add a `isDynamicFacet` method or property to `IDynamicFacet` and check if it exists?
E.g.:
**src/ui/DynamicFacet/DynamicFacetInterface.ts**
```ts
export interface IDynamicFacet extends Component, IAutoLayoutAdjustableInsideFacetColumn {
  [...]
  isDynamicFacet(): true;
}
```
**src/ui/DynamicFacet/DynamicFacet.ts**
```ts
export class DynamicFacet extends Component implements IDynamicFacet {
  [...]
  public isDynamicFacet() {
    return true;
  }
}
```
**src/ui/DynamicFacetManager/DynamicFacetManager.ts **
```ts
export class DynamicFacetManager extends Component {
  [...]
  private isDynamicFacet(component: Component) {
    return !!(component as IDynamicFacet).isDynamicFacet
  }
}
```
That would also mean that `DynamicFacetRange` would implicitly be a `DynamicFacet` since it extends `DynamicFacet` and would inherit `isDynamicFacet`.

Reply:
We could I hesitated to do something like this or keep the classes